pkg/daemons/executor/etcd.go

```diff
@@ -26,8 +26,12 @@ func (e Embedded) ETCD(args ETCDConfig) error {
 	}
 
 	go func() {
-		err := <-etcd.Err()
-		logrus.Fatalf("etcd exited: %v", err)
+		select {
+		case <-etcd.Server.StopNotify():
+			logrus.Fatalf("etcd stopped - if this node was removed from the cluster, you must backup and delete %s before rejoining", args.DataDir)
```

Comment:
so just to clarify this is for the case that the node was removed via kubectl and etcd stopped on that node, I am not sure how is that helpful if its not repeatable i.e if the etcd stopped and then it exited with a fatal error then rke2 will restart again if it ran via systemd

Reply:
If you try to rejoin a node with the same ID once it's been deleted, the joining etcd node will get an error from the cluster and then just stop. This is true regardless of whether you manually deleted the node via kubectl, or if it was deleted automatically because it failed to join. 

When you delete a node from the cluster, the ID is recorded in the raft log, and that ID can never join again. If you want to rejoin that node you have to delete the DB dir so that it generates a new ID. If you don't do this, etcd detects the error from the server, prints a warning, and then stops. This handles that for k3s; I'm not sure how to handle it for rke2 where etcd is in a pod and will just continually stop and get restarted.

Reply:
Do you think we should delete the DB dir for the user? I'm concerned that this creates the potential for data loss. Unfortunately theres no way around this - etcd doesn't allow you to rejoin a member to the cluster once it's been removed. It has to start over from scratch with a new ID and get promoted from learner again.

Reply:
no I wouldnt do that, deleting the db dir for the user would can really be dangerous, i think if its fine if it will stuck in a restart loop, I am just afraid that the user will keep keep seeing the node in a restart loop without knowing the reason, i think we should touch a file (evicted) in the db dir and check for this file and let the user know that it should be removed as well as the data dir

Reply:
When would we write this file? We don't have any direct way of detecting it since etcd just logs a message and then stops.
How would touching / checking for an `evicted` file result in any different messaging or required user action than what I'm doing here?

Reply:
Here's what you get right now after a node has been removed from the cluster (either due to stalled promote, or `kubectl delete node`):

```
{"level":"warn","ts":"2020-10-21T12:28:51.439-0700","caller":"etcdserver/server.go:1095","msg":"server error","error":"the member has been permanently removed from the cluster"}
{"level":"warn","ts":"2020-10-21T12:28:51.439-0700","caller":"etcdserver/server.go:1096","msg":"data-dir used by this member must be removed"}
{"level":"warn","ts":"2020-10-21T12:28:51.439-0700","caller":"etcdserver/server.go:2052","msg":"stopped publish because server is stopped","local-member-id":"402a826939f143ff","local-member-attributes":"{Name:centos02-0b1abeac ClientURLs:[https://10.0.1.170:2379]}","publish-timeout":"15s","error":"etcdserver: server stopped"}
{"level":"info","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/peer.go:333","msg":"stopping remote peer","remote-peer-id":"ad4df066938f2757"}
{"level":"warn","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/stream.go:301","msg":"stopped TCP streaming connection with remote peer","stream-writer-type":"unknown stream","remote-peer-id":"ad4df066938f2757"}
{"level":"warn","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/stream.go:301","msg":"stopped TCP streaming connection with remote peer","stream-writer-type":"unknown stream","remote-peer-id":"ad4df066938f2757"}
{"level":"info","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/pipeline.go:86","msg":"stopped HTTP pipelining with remote peer","local-member-id":"402a826939f143ff","remote-peer-id":"ad4df066938f2757"}
{"level":"info","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/stream.go:459","msg":"stopped stream reader with remote peer","stream-reader-type":"stream MsgApp v2","local-member-id":"402a826939f143ff","remote-peer-id":"ad4df066938f2757"}
{"level":"info","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/stream.go:459","msg":"stopped stream reader with remote peer","stream-reader-type":"stream Message","local-member-id":"402a826939f143ff","remote-peer-id":"ad4df066938f2757"}
{"level":"info","ts":"2020-10-21T12:28:51.439-0700","caller":"rafthttp/peer.go:340","msg":"stopped remote peer","remote-peer-id":"ad4df066938f2757"}
time="2020-10-21T12:28:51.443493911-07:00" level=fatal msg="etcd stopped - if this node was removed from the cluster, you must backup and delete /var/lib/rancher/k3s/server/db/etcd before rejoining"
[root@centos02 ~]#
```